lib/SqlCall.js

```diff
@@ -16,323 +16,242 @@
 // DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
-const iXml = require('./ixml');
-
 class SqlCall {
   /**
    * @description Creates a new SqlCall
    * @constructor
    */
   constructor() {
-    this.xml = iXml.iXmlNodeSqlOpen();
+    this.xml = '<sql>';
   }
 
   /**
    * @description adds sql query XML
    * @param {string} stmt
    * @param {object} [options]
    */
-  addQuery(stmt, options) {
-    if (options && options.error) {
-      this.xml += iXml.iXmlNodeSqlQueryOpen(options.error) + stmt + iXml.iXmlNodeSqlQueryClose();
-    } else {
-      this.xml += iXml.iXmlNodeSqlQueryOpen() + stmt + iXml.iXmlNodeSqlQueryClose();
-    }
+  addQuery(stmt, options = {}) {
+    this.xml += `<query error='${options.error || 'fast'}'>${stmt}</query>`;
   }
 
   /**
    * @description adds sql prepare XML
    * @param {string} stmt
    * @param {object} [options]
    */
-  prepare(stmt, options) {
-    if (options && options.error) {
-      this.xml += iXml.iXmlNodeSqlPrepareOpen(options.error) + stmt
-      + iXml.iXmlNodeSqlPrepareClose();
-    } else {
-      this.xml += iXml.iXmlNodeSqlPrepareOpen() + stmt + iXml.iXmlNodeSqlPrepareClose();
-    }
+  prepare(stmt, options = {}) {
+    this.xml += `<prepare error='${options.error || 'fast'}'>${stmt}</prepare>`;
   }
 
   /**
    * @description adds sql execute XML
    * @param {array} params
    * @param {object} [options]
    */
-  execute(params, options) {
-    if (options && options.error) {
-      this.xml += iXml.iXmlNodeSqlExecuteOpen(options.error);
-    } else {
-      this.xml += iXml.iXmlNodeSqlExecuteOpen();
-    }
+  execute(params, options = {}) {
+    this.xml += `<execute error='${options.error || 'fast'}'>`;
 
     if (params && params.length) {
       for (let i = 0; i < params.length; i += 1) {
-        if (params[i].length > 1) {
-          this.xml += iXml.iXmlNodeSqlParmOpen(params[i][1]) + params[i][0]
-          + iXml.iXmlNodeSqlParmClose();
-        } else {
-          this.xml += iXml.iXmlNodeSqlParmOpen() + params[i][0] + iXml.iXmlNodeSqlParmClose();
-        }
+        this.xml += `<parm${params[i].length ? ` io='${params[i][1]}'` : ''}>${params[i][0]}</parm>`;
```

Comment:
```suggestion
        let io = params[i].length ? `io='${params[i][1]}'` : '';
        this.xml += `<parm ${io}>${params[i][0]}</parm>`;
```
The nested format strings are very confusing. Better to hoist it out in to a separate variable. This will also have an extra space if io is not set (`<parm >`), but that's fine and it's much clearer to read.